Node Problem Detector

mburke5678 opened the pull request:
https://trello.com/c/HQfn7mca/605-document-node-adopt-node-problem-detector

mburke5678 commented:
@joelsmith I pulled down the README as a basis for the OpenShift docs on this feature. Can you take a look and advise? The docs for this feature are targeted for the current sprint and I would like to get this done by EOW.

mburke5678 commented:
@joelsmith BUMP

joelsmith commented:
Sorry that I haven't spent much time on this. Unfortunately, we still don't have our version of the NPD built, so it's hard to get much more than this in the way of specifics. I think what you have now looks good for a draft, and then we'll need to make another pass to make it match what we actually end up building when we get our own container image built.

mburke5678 commented:
Thanks for the update. Any sense on how long before work will be complete?

On Fri, Sep 15, 2017 at 12:15 AM, Joel Smith <notifications@github.com>
wrote:

> Sorry that I haven't spent much time on this. Unfortunately, we still
> don't have our version of the NPD built, so it's hard to get much more than
> this in the way of specifics. I think what you have now looks good for a
> draft, and then we'll need to make another pass to make it match what we
> actually end up building when we get our own container image built.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/5236#issuecomment-329676843>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYCpQ_s1SKJWWmSPUseH0POpdWdCGXaVks5sifnmgaJpZM4PT1ip>
> .
>



-- 

Michael Burke

Technical Writer, Customer Content Services

Red Hat

<https://www.redhat.com>

314 Littleton Rd

Westford, MA 01886

mburke@redhat.com
<https://red.ht/sig>

joelsmith commented:
@mburke5678 I thought last week with about 50% confidence that we'd be done by today, but one week later, despite our progress this week, I have about the same estimate. A lot of the uncertainty comes as a result of this being a new process (and a new-to-me process). If things go our way, we'll be done with the build stuff by the middle of next week, and then we'll have a clear picture of what things will look like in the shipped product so that we can document them.

mburke5678 commented:
@joelsmith Do we have a status on this?

joelsmith commented:
@mburke5678 sorry that I don't have this for you. We've pushed this to 3.8. For 3.7 we'll just be running it in Online for a kind of beta test.

mburke5678 commented:
@joelsmith Can you take a look at what I have so far and see if I am covering the feature right?

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,317 @@
+to overwrite the configuration.
+
+. Change the configuraton files in config/.
+
```
@joelsmith What is the `config/` directory? Do we use this in OpenShift or is it Kubernetes?

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,317 @@
++
+<1> The node selector that determines on which nodes pod replicas should be deployed
+<2> Specify the location of your cluster log directory.
+
```
@joelsmith Must this file be called `node-problem-detector.yaml` or can it be any name?

mburke5678 commented:
@joelsmith PTAL and see questions above.

mburke5678 commented:
@joelsmith PTAL and see questions above.

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,319 @@
+
+[[admin-guide-node-problem-detector-addon]]
+== Running the Node Problem Detector as an Add-on Pod
+
```
Can customers run NPD as an add-on? I don't see references in our docs on Kubernetes Add-ons, Add-on Manger. Maybe we use different wording. I can find out if we support this.

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,319 @@
+You can use a xref:../dev_guide/configmaps.adoc#dev-guide-configmaps[ConfigMap]
+to overwrite the configuration.
+
+. Exit the *_/etc/kubernetes/addons/node-problem-detector/node-problem-detector/config/kernel-monitor.json_* configuraton file.
```
@joelsmith Do we use *_/etc/kubernetes/addons/_* directory? If not where does the `kernel-monitor.json` live in OpenShift, if we use it.

joelsmith replied:
We don't use the addons framework. For ours, the user needs to edit the configmap provided by the installer:

```
oc edit configmap -n openshift-node-problem-detector node-problem-detector
```

Then edit the section for the `kernel-monitor.json` file. Changes will take effect after running pods are restarted. To restart them, it's easiest to just delete all existing pods:
```
oc delete pods -n openshift-node-problem-detector -l name=node-problem-detector
```

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,319 @@
+
+If these problems are undetected, {product-title} 
+might scheduling pods to the bad nodes.
+
```
@joelsmith How does the user access the output/reported issues?

mburke5678 commented:
@joelsmith Can you PTAL at the questions above?

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,328 @@
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json[CustomPluginMonitor]
+| On-demand (based on user configuration)
+| A custom plugin monitor for node-problem-detector to invoke and check various node problems with user defined check scripts.
+
```
@joelsmith Where are the user-defined scripts? What goes in them?

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,328 @@
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json[CustomPluginMonitor]
+| On-demand (based on user configuration)
+| A custom plugin monitor that checks various node problems with xref:admin-guide-node-problem-detector-monitor[custom, user-defined scripts].
+
```
@joelsmith I have user-defined scripts linking to my "Using Custom Monitoring Tools" section below. Is this correct?

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,328 @@
+You can use other montoring plugin tools, such as 
+link:https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/node_problem_detector_linux.go[node_problem_detector_linux.go
+].
+
```
@joelsmith I found this link to a purported custom monitor plug-in from an [Kubernetes issue on adding custom plug-ins](https://github.com/kubernetes/node-problem-detector/issues/150#issuecomment-346488233). Is this a good example?

mburke5678 commented:
@joelsmith @weinliu 
I worked in Joel's changes. I also added some of the test cases as sections:
Using NPD
Enable journald
Uninstalling NPD

Can either of you send me or point me to a NPD YAML file and Config file that I could use?

weinliu commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,509 @@
+# oc edit configmap -n openshift-node-problem-detector node-problem-detector
+----
+
+. Change the `node-problem-detector.yml` to use the configuration map:
```
Should it be 

> node-problem-detector.yaml  ?

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,511 @@
+
+[source,bash]
+----
+ansible-playbook ~/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
```
Since the local checkout location of the openshift-ansible repo will vary from person to person, I'd say that we should leave the `~/` off. Or maybe we should refer to the one installed by the openshift-ansible-playbooks RPM on a node. Here's the location of the file from the RPM if you want to use that:

```
/usr/share/ansible/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
```

joelsmith replied:
I think my earlier comment is still applicable. This should be `ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml` and the text above it should mention that the file comes from the `openshift-ansible-playbooks` RPM.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,511 @@
+      - name: config
+        configMap:
+          name: node-problem-detector-config <4>
+----
```
I'm not sure what to do here. If we want to provide a yaml file, it should probably be very close to the ones we ship for Origin and OCP.  Here's the one for Origin, but do we really want to have to keep this up to date with the right image spec?

```
apiVersion: v1
items:
- apiVersion: apps/v1
  kind: DaemonSet
  metadata:
    labels:
      app: node-problem-detector
    name: node-problem-detector
  spec:
    template:
      metadata:
        labels:                                                                                                                                                                                                                             
          name: node-problem-detector                                                                                                                                                                                                       
      spec:                                                                                                                                                                                                                                 
        containers:                                                                                                                                                                                                                         
        - command:                                                                                                                                                                                                                          
          - node-problem-detector                                                                                                                                                                                                           
          - --logtostderr                                                                                                                                                                                                                   
          - --system-log-monitors=/etc/npd/kernel-monitor.json,/etc/npd/docker-monitor.json                                                                                                                                                 
          env:                                                                                                                                                                                                                              
          - name: NODE_NAME                                                                                                                                                                                                                 
            valueFrom:                                                                                                                                                                                                                      
              fieldRef:                                                                                                                                                                                                                     
                apiVersion: v1                                                                                                                                                                                                              
                fieldPath: spec.nodeName                                                                                                                                                                                                    
          image: docker.io/openshift/node-problem-detector:v3.10.0                                                                                                                                                                           
          imagePullPolicy: IfNotPresent                                                                                                                                                                                                     
          name: node-problem-detector                                                                                                                                                                                                       
          resources: {}                                                                                                                                                                                                                     
          securityContext:                                                                                                                                                                                                                  
            privileged: true                                                                                                                                                                                                                
          terminationMessagePath: /dev/termination-log                                                                                                                                                                                      
          terminationMessagePolicy: File                                                                                                                                                                                                    
          volumeMounts:                                                                                                                                                                                                                     
          - mountPath: /host/log                                                                                                                                                                                                            
            name: log                                                                                                                                                                                                                       
            readOnly: true                                                                                                                                                                                                                  
          - mountPath: /etc/localtime
            name: localtime
            readOnly: true
          - mountPath: /etc/npd
            name: config
        restartPolicy: Always
        securityContext: {}
        serviceAccount: node-problem-detector
        serviceAccountName: node-problem-detector
        terminationGracePeriodSeconds: 30
        volumes:
        - hostPath:
            path: /var/log/
            type: ""
          name: log
        - hostPath:
            path: /etc/localtime
            type: ""
          name: localtime
        - configMap:
            name: node-problem-detector
          name: config
    updateStrategy:
      type: RollingUpdate
```

joelsmith commented:
Here's the default config map that we ship and install with NPD.

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: node-problem-detector
data:
  docker-monitor.json: |
    {
        "plugin": "journald",
        "pluginConfig": {
                "source": "docker"
        },
        "logPath": "/host/log/journal",
        "lookback": "5m",
        "bufferSize": 10,
        "source": "docker-monitor",
        "conditions": [],
        "rules": [
                {
                        "type": "temporary",
                        "reason": "CorruptDockerImage",
                        "pattern": "Error trying v2 registry: failed to register layer: rename /var/lib/docker/image/(.+) /var/lib/docker/image/(.+): directory not empty.*"
                }
        ]
    }
  kernel-monitor.json: |
    {
        "plugin": "journald",
        "pluginConfig": {
                "source": "kernel"
        },
        "logPath": "/host/log/journal",
        "lookback": "5m",
        "bufferSize": 10,
        "source": "kernel-monitor",
        "conditions": [
                {
                        "type": "KernelDeadlock",
                        "reason": "KernelHasNoDeadlock",
                        "message": "kernel has no deadlock"
                }
        ],
        "rules": [
                {
                        "type": "temporary",
                        "reason": "OOMKilling",
                        "pattern": "Kill process \\d+ (.+) score \\d+ or sacrifice child\\nKilled process \\d+ (.+) total-vm:\\d+kB, anon-rss:\\d+kB, file-rss:\\d+kB"
                },
                {
                        "type": "temporary",
                        "reason": "TaskHung",
                        "pattern": "task \\S+:\\w+ blocked for more than \\w+ seconds\\."
                },
                {
                        "type": "temporary",
                        "reason": "UnregisterNetDevice",
                        "pattern": "unregister_netdevice: waiting for \\w+ to become free. Usage count = \\d+"
                },
                {
                        "type": "temporary",
                        "reason": "KernelOops",
                        "pattern": "BUG: unable to handle kernel NULL pointer dereference at .*"
                },
                {
                        "type": "temporary",
                        "reason": "KernelOops",
                        "pattern": "divide error: 0000 \\[#\\d+\\] SMP"
                },
                {
                        "type": "permanent",
                        "condition": "KernelDeadlock",
                        "reason": "AUFSUmountHung",
                        "pattern": "task umount\\.aufs:\\w+ blocked for more than \\w+ seconds\\."
                },
                {
                        "type": "permanent",
                        "condition": "KernelDeadlock",
                        "reason": "DockerHung",
                        "pattern": "task docker:\\w+ blocked for more than \\w+ seconds\\."
                }
        ]
    }
```

To edit, somebody can do:

```
oc edit -n openshift-node-problem-detector configmap node-problem-detector
```

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,376 @@
+Standalone mode uses an insecure API server connection.
+====
+
+[[admin-guide-node-problem-detector-gce]]
```
I think we ought to remove this entire section. It's Kube-specific and not really relevant to OpenShift.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,376 @@
+
+[NOTE]
+====
+The node problem detector does not detect problems reported by log tools, such as journald or rsyslogd.
```
I'm not sure what this note is trying to say, but we should probably either re-word it (depending upon what it's trying to say) or remove it. NPD definitely supports reading log entries from journald and can be configured to consume log files written by rsyslogd.

mburke5678 replied:
@joelsmith Removing. I wanted to ask you about this. Got this from upstream. Did I read it wrong?
"The kernel issue detection of node problem detector only supports file based kernel log now. It doesn’t support log tools like journald."
https://kubernetes.io/docs/tasks/debug-application-cluster/monitor-node-health/#limitations

joelsmith replied:
You read it right, but their docs are old compared to the code. NPD has supported journald upstream for at least a year.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,376 @@
+
+[source,bash]
+----
+# oc node-problem-detector --system-log-monitors <path to config files>
```
This looks like a typo.  The `oc` should be removed and we should probably add a note that the `node-problem-detector` command line can be modified in the daemonset to add the `--system-log-monitors` switch. Basically, the method to configure this option is like the above one where the `oc edit ds` is used to modify the container's command. In fact, perhaps this section could be combined with the one above it since they're covering the configuration of the same thing.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,376 @@
+
+[NOTE]
+====
+The add-on manager does not support configuration maps. If you are running the node problem detector running as cluster add-on, 
```
We can remove this note since OpenShift isn't making use of the add-on manager for NPD.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,376 @@
+        "plugin": "journald",""
+----
+
+<1> Sends the output to standard output (stdout).
```
I think this line and the next few were left around from a previous revision.

mburke5678 commented:
@joelsmith PTAL

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,426 @@
++
+[source,bash]
+----
+oc edit ds -n openshift-node-problem-detector configmap node-problem-detector
```
Sorry if I told you wrong before. This one should not have `ds` in it. This command is to edit the `configmap` (which comes later in the command).

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,426 @@
+ansible-playbook ~/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
+----
+
+You can also manually create the daemonset object, using the following steps. 
```
We either need to 1) take the manual install section out or 2) update it to match the `ds` we create with the installer. My preference is 1 since otherwise we'd need to update this doc for every version to point to the correct NPD image for the product version for the doc. (i.e. origin vs. ocp, 3.9 vs 3.10, etc). If you want to keep it, let me know and I'll send you the `ds` from the installer.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,426 @@
+----
+
+[[admin-guide-node-problem-detector-run]]
+== Reinstalling the Node Problem Detector
```
This section is called "Reinstalling", but I don't think we ever tell how to install it (excepting the manual install section below). The "reinstall" instructions work for an "install" too, or we can tell them to set the inventory variable `openshift_node_problem_detector_install=true` prior to running the OpenShift installer. Maybe that belongs in this section https://docs.openshift.com/container-platform/3.9/install_config/install/advanced_install.html#configuring-ansible or elsewhere. I'm not familiar with our docs conventions.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,426 @@
+[source,bash]
++
+----
+oc edit ds -n <project> configmap <node-problem-detector-name>
```
Sorry if I told you wrong before. This one should not have `ds` in it. This command is to edit the configmap (which comes later in the command).

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,426 @@
+For example:
++
+----
+oc edit ds -n openshift-node-problem-detector configmap node-problem-detector
```
Sorry if I told you wrong before. This one should not have `ds` in it. This command is to edit the `configmap` (which comes later in the command).

If you prefer to have the `-n <project>` later in the command, then this should read
```
oc edit configmap -n openshift-node-problem-detector node-problem-detector
```
or if you prefer the ns first (like the previous version), it would look like:
```
oc edit -n openshift-node-problem-detector configmap node-problem-detector
```

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,428 @@
+----
+[OSEv3:vars]
+openshift_node_problem_detector_state=absent
+----
```
@joelsmith Should this be openshift_node_problem_detector_install=**false**

joelsmith replied:
No, it's correct as written.

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,511 @@
+| Problem Daemon | NodeCondition | Description
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json[KernelMonitor]
+| KernelDeadlock
```
@joelsmith Here, the KernelDeadlock is a NodeCondition reported by the KernelMonitor daemon. A bit hard to see in this format.
| Problem Daemon | NodeCondition  | Description
|KernelMonitor       | KernelDeadlock | A system log that monitors kernel logs and reports problem according to predefined rules.

mburke5678 commented:
@joelsmith I made some changes. 
* I moved the example daemonset higher and added callouts
* Changed the Using NPD heading to "Verifying the NPD is Running: and added steps to issue fake error message.
* Added callouts to the section on adding the `alsologtostderr=true` variable, 

Questions:
Do I have `alsologtostderr=true` in the right location in the file?
Do we need to add  ` --system-log-monitors=/etc/npd/kernel-monitor.json,/etc/npd/docker-monitor.json`?
I watched your demo again and you suggested that NPD read the simulated error message and patched the object. What do you mean by patching the object? 
Do we need to tell users how to integrate the controllers that fix the errors discovered by NPD?

mburke5678 commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,381 @@
+
+The xref:../install/index.adoc#install-planning[installation] creates 
+a Node Problem Detector daemonset by default and creates a project for the detector, called `openshift-node-problem-detector`.
+
```
@joelsmith Does the user need to do anything on master and/or on the nodes to start NPD? oc create the daemonset? Or, does it run by default?

joelsmith replied:
No, if they have run the installer with the variable `openshift_node_problem_detector_install=true` on the inventory, then the daemonset will be configured and the cluster will run NPD for them.  However, I don't see the installation via openshift-ansible documented anywhere.  Could we direct users to know that they need to set `openshift_node_problem_detector_install` to `true` for their install?

mburke5678 replied:
Which section does `openshift_node_problem_detector_install=true` in the inventory file?

mburke5678 commented:
@joelsmith How is this topic looking; would love to get over to QE as soon as we get it right.

joelsmith commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,391 @@
+        command:
+        - node-problem-detector
+        - --alsologtostderr=true <1>
+          --log_dir="/tmp" <2>
```
This line needs a dash in the same column as the rows above and below it. The `command` parameter expects a list of strings, and each item in the list in YAML is designated by a single dash before it. Sorry I didn't notice before.

joelsmith commented:
> Questions:
> Do I have alsologtostderr=true in the right location in the file?

It looks good to me.

> Do we need to add --system-log-monitors=/etc/npd/kernel-monitor.json,/etc/npd/docker-monitor.json?

We ship with them enabled by default, but people configuring NPD for themselves would need to add them (or add their own custom config files) to the command line. I think that the way you have it is good. 

> I watched your demo again and you suggested that NPD read the simulated error message and patched the object. What do you mean by patching the object?

"Patching an object" just refers to the way that an API client modifies an Kubernetes API object. So what I meant was that NPD updates the node object's `conditions` field with the current state of any conditions that it manages.

> Do we need to tell users how to integrate the controllers that fix the errors discovered by NPD?

No, because currently no such controllers exist (at least no publicly available controllers exist). I think the idea is that NPD is a building block and we'll "get there" eventually.

weinliu commented:
@mburke5678 I've went through the updated doc, LGTM, no further comments

mburke5678 commented:
@joelsmith One last (I hope) question. if the NPD is not installed in a cluster, the user should?

. Edit the *_/etc/ansible/hosts_* inventory file.

. Set the `openshift_node_problem_detector_install` to `true`.

. Reinstall {product-title}.

mburke5678 commented:
@openshift/team-documentation PTAL

ahardin-rh commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,402 @@
+The Node Problem Detector reads system logs and watches for specific entries and makes these problems visible to the control plane,
+which you can view using {product-title} commands, such as `oc get node` and `oc get event`. 
+You could then take action to correct these problems as appropriate or capture the messages using a tool of your choice, 
+such as the {product-title} xref:..//security/monitoring.adoc#security-monitoring[log monitoring].
```
looks like an extra / in the file path

ahardin-rh commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,402 @@
+== Example Node Problem Detector Output
+
+The following examples show output from the Node Problem Detector watching for kernel deadlock node condition on a specific node.
+The command uses `oc get node` to watch a specific node filtering for a `KernelDeadlock` entry in a log. 
```
Strange line break

ahardin-rh commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,402 @@
+The Node Problem Detector consumes resources. If you use the Node Problem Detector, make sure you have enough nodes to balance cluster performance.
+====
+
+== Installing the Node Problem Detector
```
Missing section ID

ahardin-rh commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,402 @@
+You must manually change the parameter to `true` when installing the Node Problem Detector.
+====  
+
+If xref;admin-guide-node-problem-detector-verify[the Node Problem Detector is not installed]: 
```
xref:  (need a colon instead of semicolon)

ahardin-rh commented on admin_guide/node_problem_detector.adoc:
```diff
@@ -0,0 +1,402 @@
++
+<1> Sends the output to standard output (stdout).
+<2> Path to the error log.
+<3> Comma-separated path to the plugin configuration files.
```
plug-in

ahardin-rh commented:
@mburke5678 Just a few comments from me. :beers:

joelsmith commented:
> @joelsmith One last (I hope) question. if the NPD is not installed in a cluster, the user should?
> 
> . Edit the /etc/ansible/hosts inventory file.
> 
> . Set the openshift_node_problem_detector_install to true.
> 
> . Reinstall {product-title}.

If everything is already installed and they just want to add NPD, they don't need to set any inventory vars or anything, they should just run `playbooks/openshift-node-problem-detector/config.yml`.

mburke5678 commented:
@joelsmith Does the user need to set `openshift_node_problem_detector_install=true`

```
If xref:admin-guide-node-problem-detector-verify[the Node Problem Detector is not installed]:

. Add following options in Ansible inventory file:
+
[source,yaml]
----
[OSEv3:vars]
openshift_node_problem_detector_install=true
----

. Run the *openshift-node-problem-detector/config.yml* playbook: 

----
# ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/openshift-node-problem-detector/config.yml
----

```

joelsmith commented:
@mburke5678 no, not if they run that playbook directly. That variable controls whether or not the NPD playbook is called via the optional components playbook. If running the NPD playbook directly, no variable needs to be set.
